hw/ip/sram_ctrl/doc/dv/index.md

```diff
@@ -86,10 +136,62 @@ The following covergroups have been developed to prove that the test intent has
 #### Scoreboard
 The `sram_ctrl_scoreboard` is primarily used for end to end checking.
 It creates the following analysis ports to retrieve the data monitored by corresponding interface agents:
-* analysis port1:
-* analysis port2:
+* tl_a_chan_fifo: TL address channel for CSR accesses
+* tl_d_chan_fifo: TL data channel for CSR accesses
+* alert_fifos: Alert handshakes
+* sram_tl_a_chan_fifo: TL address channel for memory accesses
+* sram_tl_d_chan_fifo: TL data channel for memory accesses
+* kdi_fifo: For key refresh operations from OTP
 <!-- explain inputs monitored, flow of data and outputs checked -->
 
+All CSR accesses made to the SRAM_CTRL register file are tracked and predicted accordingly inside the scoreboard.
+
+Verifying memory accesses is a little more involved of a process, and makes
+heavy use of the [`mem_bkdr_if`]({{< relref "hw/dv/sv/mem_bkdr_if/mem_bkdr_if.sv" >}}).
+
+At a high level, whenever a TL access is made to memory, all relevant information is stored in a transaction-level struct
+(such as write_enable, address, write data, mask, key, and nonce).
+
+Depending on the transaction type (read or write), we then hold the transaction
+in limbo until it will be reflected in the underlying memory macro.
+For example, all read transactions will return a response in 1 cycle, but a write may not be reflected in memory until many cycles later if it is immediately followed by a read request to the same address.
+This data forwarding is modeled in the scoreboard such that we check all memory transactions
+in the exact same order that they are actually executed by the RAM primitive.
+
+At this time, we use the memory backdoor interface to read the word in memory at the transaction's address
+and then compare it against the requested write data or the read response data,
+using an appropriate byte-mask depending on the granularity of the memory access.
+
+We use a specific set of methods in the memory backdoor interface that utilize
+the memory scrambling golden model in order to provide the testbench with the
+correct unencrypted data for these comparisons.
+
+For example, say we perform a memory access to address `0x0` to write a `1` into memory.
+Internally, the scrambling RAM primitive will map address `0x0` to some other address,
+say scrambled address `0x4`, and will also encrypt the data.
+In the scoreboard we detect this transaction on the TLUL bus and use the memory backdoor interface
+and scrambling golden model to perform a backdoor read to logical address `0x1`.
```

Comment:
I don't understand: the requested access to address 0 is scrambled to address 4, and then you read from address 1? Could you clarify?

Reply:
whoops! this should be `0x0`, good catch.